spirv_msl.cpp

```diff
@@ -1276,6 +1432,38 @@ string CompilerMSL::argument_decl(const SPIRFunction::Parameter &arg)
 	return join(constref ? "const " : "", type_to_glsl(type), "& ", to_name(var.self), type_to_array_glsl(type));
 }
 
+// If we're currently in the entry point function, and the object
+// has a qualified name, use it, otherwise use the standard name.
+string CompilerMSL::to_name(uint32_t id, bool allow_alias)
+{
+    if (current_function && (current_function->self == entry_point) ) {
+        string qual_name = meta.at(id).decoration.qualified_alias;
+        if ( !qual_name.empty() )
+            return qual_name;
+    }
+    return Compiler::to_name(id, allow_alias);
+}
+
+// Returns a name that combines the name of the struct with the name of the member
+string CompilerMSL::to_qualified_member_name(const SPIRType &type, uint32_t index)
+{
+    // Get name and strip any underscore prefix
+    string mbr_name = to_member_name(type, index);
+    size_t startPos = mbr_name.find_first_not_of("_");
+    mbr_name = (startPos != std::string::npos) ? mbr_name.substr(startPos) : "";
+    return join(to_name(type.self), "_", mbr_name);
+}
+
+// Ensures that the specified struct member name is permanently usable by prepending
+// an alpha char if the first chars are _ and a digit, which indicate a transient name.
+string CompilerMSL::ensure_member_name(string mbr_name)
```

Comment:
spirv_msl creates dynamic structs to convert global vars to function arg structs. The name of global is used for struct member. However, if no alpha name is provided and default numeric name is used, when emitting the struct, emit_struct() clears the numeric names and regenerates based on member position in struct. The original global ID form is lost. This causes trouble maintaining references between struct member declarations and accesses. Forcing an alpha name preserves the original numeric names.